### PR TITLE
Fix spec RTL12

### LINKED ISSUE #1271: Fix no-connection-serial RTL12
* Fix RTL12 condition for presence channel attach https://github.com/ably/ably-dotnet/blob/c2a514ad2cc9401547c8c6ccb7c23b87101131a5/src/IO.Ably.Shared/Realtime/ChannelMessageProcessor.cs#L73
* Fix internal presence members enter with id
* Fix code as per latest jan 12 commits https://github.com/ably/ably-go/pull/613/commits

┆Issue is synchronized with this [Jira Task](https://ably.atlassian.net/browse/ECO-4276) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by sacOO7**
- Fixed #1271 
- Fixes part of presence and no-connection-serial ( as discussed during review of ably-go and ably-cocoa )

### CODE REVIEW

File: src/IO.Ably.Shared/Extensions/PresenceExtensions.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public static bool IsNewerThan(this PresenceMessage thisMessage, PresenceMessage
             // RTP2b1
             if (thisMessage.IsSynthesized() || thatMessage.IsSynthesized())
             {
-                return thisMessage.Timestamp > thatMessage.Timestamp;
+                return thisMessage.Timestamp >= thatMessage.Timestamp;
```
Comment (ttypic) [inline src/IO.Ably.Shared/Extensions/PresenceExtensions.cs:16]: could we rename `thisMessage` and `thatMessage` to align with the spec, spec mentions incoming message and existing message?

File: src/IO.Ably.Shared/Extensions/PresenceExtensions.cs [hunk 2]
```diff
@@ -35,7 +35,7 @@ public static bool IsNewerThan(this PresenceMessage thisMessage, PresenceMessage
 
             if (msgSerialThis == msgSerialThat)
             {
-                return indexThis > indexThat;
+                return indexThis >= indexThat;
```
Comment (ttypic) [inline src/IO.Ably.Shared/Extensions/PresenceExtensions.cs:38]: Spec doesn't cover situation when indexes are equal, I believe we should update the spec first if it's correct behavior

**Review by ttypic — CHANGES_REQUESTED** [review]

File: src/IO.Ably.Shared/Extensions/PresenceExtensions.cs (hunk 1 above)
Comment (sacOO7) [reply src/IO.Ably.Shared/Extensions/PresenceExtensions.cs:16]: Yeah, sounds great !
This was old implementation, so I didn't try to change it as such. We have changed lot of things in ably-go

File: src/IO.Ably.Shared/Extensions/PresenceExtensions.cs (hunk 2 above)
Comment (sacOO7) [reply src/IO.Ably.Shared/Extensions/PresenceExtensions.cs:38]: Sure

File: src/IO.Ably.Shared/Extensions/PresenceExtensions.cs (hunk 1 above)
Comment (sacOO7) [reply src/IO.Ably.Shared/Extensions/PresenceExtensions.cs]: Okay, I have made it verbose and seems change wasn't really needed 
https://github.com/ably/ably-dotnet/pull/1275/commits/5c77955e253355fe5beb79ae3ed2b7a126379622

**Review by ttypic — APPROVED** [review]
LGTM

File: src/IO.Ably.Shared/Extensions/PresenceExtensions.cs (hunk 1 above)
Comment (sacOO7) [reply src/IO.Ably.Shared/Extensions/PresenceExtensions.cs]: Thanks !!

File: src/IO.Ably.Shared/Extensions/PresenceExtensions.cs (hunk 2 above)
Comment (sacOO7) [reply src/IO.Ably.Shared/Extensions/PresenceExtensions.cs]: Thanks !!

**Pull request merged**